### PR TITLE
Synonym sync: Abbreviation exclusions

### LINKED ISSUE #652: Synonym sync: Abbreviation exclusions
## Overview
We have a rule (e.g. uppercase, <10 chars) for determining when a synonym is an abbreviation. However, there are / will be specific cases that we know of where a synonym may fall under this logic but in fact not be an abbreviation. For those, we will read in a specific config file and make sure that any synonyms matching the strings in those configs will not be marked as an abbreviation.

## Sub-tasks
- [x] 1. Set up code to utilize the config
  - For now can make a mock config file
- [x] 2. Add and populate abbreviation exclusions config file

### PR DESCRIPTION
**Pull request by joeflack4**
Resolves #652

## Overview
Synonym sync: Abbreviation exclusions
- Added abbreviation exclusion rules to config
- Renamed `mondo-exclusion-values.yml` --> `mondo-exclusion-configs.yml`: I didn't rename it just mondo-exclusions in order to disambiguate from similarly named SOURCE_exclusions files, which serve the purpose of excluding entire terms. I went with the general 'configs' name, because these exclude a variety of different things now, from exclusion of entire synonyms being added, to exclusion of synonym type axioms.

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [x] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [ ] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

---

[Google sheet](https://docs.google.com/spreadsheets/d/1ftEZ47-JDe-1Z_fRWyO5NS39qT_-8MZsHXHEKz9aZ9Q/edit?gid=978570179#gid=978570179)

### CODE REVIEW

**Comment by twhetzel** [comment]
@joeflack4 what goal was run to generate the data files in this PR?

File: src/ontology/reports/sync-synonym/synonym_sync_combined_cases.tsv
Comment (joeflack4) [inline src/ontology/reports/sync-synonym/synonym_sync_combined_cases.tsv:1]: > what goal was run to generate the data files in this PR?

`sh run.sh make sync-synonyms -B`
That is done with certain prereqs commented out (e.g. components) during the interim. This is a routine Nico and I have, assuming that freshly updated sources are not a necessary part of seeing the effects of the PR.

File: src/scripts/sync_synonym.py [hunk 1]
```diff
@@ -68,14 +68,14 @@ def _query_synonyms(ids: List[CURIE], db: SqlImplementation) -> pd.DataFrame:
...
-def _add_synonym_type_inferences(row: pd.Series) -> str:
+def _add_synonym_type_inferences(row: pd.Series, exclusions: Set[str]) -> str:
     """Append additional synonym_type(s) based on logical rules."""
     syn: str = row['synonym']
     syn_type_str: str = row['synonym_type']
     types: Set[URI] = set(syn_type_str.split('|') if syn_type_str else [])
 
     # Acronym: uppercase, no numbers, no whitespace, <10 chars
-    if syn.isupper() and not any(char.isspace() for char in syn) and len(syn) < 10:
+    if syn not in exclusions and syn.isupper() and not any(char.isspace() for char in syn) and len(syn) < 10:
```
Comment (joeflack4) [inline src/scripts/sync_synonym.py:78]: > Did you confirm from the curated examples that the list of synonyms that should not be tagged as an abbreviation are not tagged now?

Yep! I checked a few cases in the new [v8 google sheet](https://docs.google.com/spreadsheets/d/1ftEZ47-JDe-1Z_fRWyO5NS39qT_-8MZsHXHEKz9aZ9Q/edit?gid=978570179#gid=978570179), and they're longer being tagged as such.

File: src/ontology/reports/sync-synonym/synonym_sync_combined_cases.tsv
Comment (twhetzel) [reply src/ontology/reports/sync-synonym/synonym_sync_combined_cases.tsv:1]: sounds good

File: src/scripts/sync_synonym.py (hunk 1 above)
Comment (twhetzel) [reply src/scripts/sync_synonym.py:78]: Thanks for the link to the data file. I checked all the ones in the [curated file](https://docs.google.com/spreadsheets/d/1y5jCmHN51ydBOcdqmGTBHjDeOHTPuYWd165qEYnNtIk/edit?gid=0#gid=0) and none are tagged as abbreviations now as expected with this PR.

**Review by twhetzel — APPROVED** [review]
Looks good! All synonyms that should be excluded from being tagged as an abbreviation are excluded.

**Pull request merged**